application/comit_node/src/comit_client/bam.rs

```diff
@@ -88,11 +88,27 @@ impl Client for BamClient {
                 Ok(response) => match response.status() {
                     Status::OK(_) => {
                         info!("{} accepted swap request: {:?}", socket_addr, response);
-                        match serde_json::from_value(response.body().clone()) {
+                        match serde_json::from_value(response.get_body().clone()) {
                             Ok(response) => Ok(Ok(response)),
                             Err(_e) => Err(SwapResponseError::InvalidResponse),
                         }
                     }
+                    Status::SE(20) => {
+                        info!("{} declined swap request: {:?}", socket_addr, response);
+                        Ok(Err({
+                            let reason = response
+                                .get_header("REASON")
```

Comment:
I would appreciate it if we can discuss changes to the COMIT protocol first before they are implemented!
ATM, it is completely undocumented. We should flesh out first, how we want things to be before they are implemented.

Regarding this particular change: I don't see a usecase for custom messages provided by the user. Those are only an attack vector for spamming or possible exploits through buffer overflows in other implementations.

If we want to provide reasons for rejections, I would suggest to discuss and specify a limited set of possible reasons (BAD_RATE for example). Everything else is useless in regards to interoperability and automation which is what we are striving for.

PS: I'll try not to work from now on :)

Reply:
This was just my interpretation of this comment: `// TODO: Implement SwapReject::Decline(reason)`.

But I completely agree with you. I have decided instead to keep the field as an enum, which is something that I documented in one of the commit messages as a future improvement (and what you suggest). For now the enum only has one variant `Unspecified`.

Thank you for the comment, but please do stop working :laughing:

Reply:
"I would appreciate it if we can discuss changes to the COMIT protocol first before they are implemented!" -> you do understand that we may do changes to the protocol while you are away? 😇

Reply:
Having written the `TODO` comment and had some verbal discussions with the team I realize that I did not express myself clearly enough.

I was proposing to actually have a range of Error code `Status::SE` that could express defined/common reason of rejections such as:
- `RateDeclined`
- `UnsupportedLedger`
- `InternalError`
- `MissingLiquidity` (this is an idea, not feasible at this stage).

Do we agree that `Status::SE` is to be defined by the application level (ie, COMIT being an application level for BAM).

Reply:
@D4nte The rfc says that `SE20`  and above are reserved for the application.

Reply:
ok cool, so I guess we can use those for COMIT :)